Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1358,7 +1358,7 @@ public boolean cantBeAttachedBy(MageObject attachment, Ability source, Game game
                 && attachmentPermanent != null
                 && attachmentPermanent.getSpellAbility() != null
                 && !attachmentPermanent.getSpellAbility().getTargets().isEmpty()) {
-            canAttach = attachmentPermanent.getSpellAbility().getTargets().get(0).canTarget(attachmentPermanent.getControllerId(), this.getId(), source, game);
+            canAttach = attachmentPermanent.getSpellAbility().getTargets().get(0).withNotTarget(true).stillLegalTarget(attachmentPermanent.getControllerId(), this.getId(), source, game);
```

Comment:
looks like `withNotTarget` mutates the original target here? That's a bad side effect. Need to protect with `copy()` first if going that route.

Also, adding some comments here and a reference to the issue #9583 would be good.